modules/sw360/sw360-workflow/src/test/java/org/eclipse/sw360/antenna/sw360/workflow/processors/SW360EnricherTest.java

```diff
@@ -101,15 +107,38 @@ SW360MetaDataReceiver createMetaDataReceiver(Map<String, String> configMap) {
         return configMap;
     }
 
+    @Test
+    public void testDefaultConnectionFactoryIsCreated() {
+        assertThat(sw360Enricher.getConnectionFactory()).isNotNull();
+    }
+
     @Test
     public void testMetaDataReceiverIsCreated() {
-        sw360Enricher = new SW360Enricher();
+        SW360ConnectionConfigurationFactory connectionFactory = mock(SW360ConnectionConfigurationFactory.class);
+        HttpClient httpClient = mock(HttpClient.class);
+        ObjectMapper mapper = mock(ObjectMapper.class);
+        SW360Connection connection = mock(SW360Connection.class);
+        when(antennaContextMock.getHttpClient()).thenReturn(httpClient);
+        when(antennaContextMock.getObjectMapper()).thenReturn(mapper);
+        when(connectionFactory.createConnection(any(), eq(httpClient), eq(mapper))).thenReturn(connection);
+
+        sw360Enricher = new SW360Enricher(connectionFactory);
         sw360Enricher.setAntennaContext(antennaContextMock);
 
         SW360MetaDataReceiver metaDataReceiver = sw360Enricher.createMetaDataReceiver(createStandardConfigMap());
         assertThat(metaDataReceiver).isNotNull();
-        verify(toolConfigMock).getProxyHost();
-        verify(toolConfigMock).getProxyPort();
+
+        @SuppressWarnings("unchecked")
+        ArgumentCaptor<SW360ConnectionConfigurationFactory.Getter<String>> captor =
+                ArgumentCaptor.forClass(SW360ConnectionConfigurationFactory.Getter.class);
+        verify(connectionFactory).createConnection(captor.capture(), eq(httpClient), eq(mapper));
+        SW360ConnectionConfigurationFactory.Getter<String> getter = captor.getValue();
+        for (Map.Entry<String, String> e : createStandardConfigMap().entrySet()) {
+            assertThat(getter.apply(e.getKey())).isEqualTo(e.getValue());
```

Comment:
could this ever assert to false? 
and if you are testing the getter.apply method, is this the right place?

Reply:
What is tested here is whether a correct getter for configuration settings has been passed to the connection factory.  The factory uses this getter to read the configuration properties. So it is checked whether these read operations yield the expected values.

I don't know why this mechanism with the Getter objects is used; I did not change this.